packages/toolkit/src/utils/appValidation.ts

```diff
@@ -30,7 +34,21 @@ function findFunctionSignatureCollisions(abi1: AbiItem[], abi2: AbiItem[]) {
   return collisions
 }
 
-export function checkSignatureCollisionsWithProxy(abi: AbiItem[]) {
+function checkAssignedGlobalVariables(sourceCode: string) {
+  const assignments = parseGlobalVariableAssignments(sourceCode)
+  const assignments2 = parseConstructorAssignments(sourceCode)
+  if (assignments.length > 0 || assignments2.length > 0) {
+    console.log(
```

Comment:
I think it would be preferable that the toolkit doesn’t log to the console since it is a library.

Reply:
Definitely there should not be a console.log @macor161 said. I would recommend to either throw or return a warnings data structure

Reply:
Added throw declarations.